### PR TITLE
Move website build

### PR DESCRIPTION
**Pull request by landinjm**
# Description
Added github actions to build doxygen documentation and move it to gh-pages.

### CODE REVIEW

**Comment by fractalsbyx** [comment]
Is there a way that I can preview the site?

**Comment by landinjm** [comment]
Pull branch and follow instructions in doc/README.md

You can open up doc/doxygen/html/index.html in any browser to view the website.

I didn't spend much time making the site look pretty, so if you have suggestions please list them here

File: doc/doxygen/user_manual/installation/install_prismspf.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+## Traditional Installation (Non-Docker)
+### Downloading PRISMS-PF
+PRISMS-PF is available for download at our [GitHub site](https://github.com/prisms-center/phaseField). The recommended method for downloading the PRISMS-PF source code is through git. Using a Linux/Unix terminal, go to the directory where you want PRISMS-PF to be located. To clone the repository, type:
+```
+$ git clone https://github.com/prisms-center/phaseField.git
+```
+Git will then download the PRISMS-PF source code into a directory entitled ''phaseField''. A resource for learning to use Git can be found [here](https://git-scm.com/book).
+
+If you prefer not to use Git, a zip file containing the PRISMS-PF source code can be downloaded [here](https://github.com/prisms-center/phaseField/releases).
+
```
Comment (fractalsbyx) [inline doc/doxygen/user_manual/installation/install_prismspf.md:12]: Add instructions to checkout latest version.

File: doc/doxygen/user_manual/running_apps/running_apps.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+- precipiateEvolution_pfunction: Like precipitateEvolution, but loads inputs using PRISMS IntegrationTools. (2D)
+- MgNd_precipitate_single_Bppp: Similar to precipiateEvolution, but uses the KKS model rather than the WBM model for the free energy functional. The parameters are set for \f$\beta\f$''' precipitates in an Mg-Nd alloy from [this publication](https://www.sciencedirect.com/science/article/pii/S1359645417305281). (3D)
+- nucleationModel: KKS precipitation model that makes use of the PRISMS-PF explicit nucleation capabilities. (2D)
+- nucleationModel_preferential: Like nucleationModel, but with a zone with an increased nucleation rate to simulate a grain boundary. (2D)
+
+A directory for each of these apps can be found in the [applications directory](https://github.com/prisms-center/phaseField/tree/master/applications) (i.e. phaseField/applications). The apps contain a formulation file giving the governing equations. In addition to the 24 apps listed above, some app names may be preceded by an underscore. The underscore is used to denote apps that are still under active development.
+
+## Compiling and Running the Allen-Cahn Example App
```
Comment (fractalsbyx) [inline doc/doxygen/user_manual/running_apps/running_apps.md:32]: At some point before this, hyperlink to install_prismspf

File: doc/doxygen/user_manual/running_apps/running_apps.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+## Running the Other Example Applications
+Running the other example applications is as simple as going to the directory for the application of interest and repeating the steps in [the Allen-Cahn section above](#Compiling-and-Running-the-Allen-Cahn-Example-App). For example, to run the Cahn-Hilliard example application, when you are currently in the directory for the Allen-Cahn example application, you would type the following commands:
+```
+$ cd ../cahnHilliard/
+$ cmake .
+$ make debug
+$ mpirun -n 1 main
+```
+
+## Running in Release Mode and with Multple Processors
```
Comment (fractalsbyx) [inline doc/doxygen/user_manual/running_apps/running_apps.md:124]: Release is default now?

File: doc/doxygen/user_manual/app_structure/app_structure.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- formulation.pdf
+- CMakeLists.txt
+
+It may contain two other files:
+- postprocess.cc
+- nucleation.cc
+
+The following section [The Parameters File](#input_file) will discuss the structure and options for the input file, parameters.prm. Users who do not need to change the governing equations or make other fundamental changes to the application will only need to modify this file.
+
+[The App Files](#app_files) discusses the structure of the files that define an application: equations.cc, ICs_and_BCs.cc, postprocess.cc, nucleation.cc, customPDE.h. Users who want to substantially modify existing PRISMS-PF applications or create their own will find this section useful.
```
Comment (fractalsbyx) [inline doc/doxygen/user_manual/app_structure/app_structure.md:18]: customPDE.h should be first or second in this list

File: doc/doxygen/user_manual/custom_apps/custom_apps.md
Comment (fractalsbyx) [inline doc/doxygen/user_manual/custom_apps/custom_apps.md:1]: add a small section outlining what to change in cmakelists if additional .cc files are added (something like add_executable(main.cc customfile.cc))

File: doc/doxygen/user_manual/running_apps/running_apps.md (hunk 2 above)
Comment (fractalsbyx) [reply doc/doxygen/user_manual/running_apps/running_apps.md:32]: I haven't visualized the site yet. Im guessing this already exists in a sidebar.

File: doc/doxygen/landing_page.md
Comment (fractalsbyx) [inline doc/doxygen/landing_page.md:1]: May be redundant to a link on the sidebar, but I would suggest a big hyperlink labeled "Getting Started" that links to the install page.

File: doc/doxygen/user_manual/app_files/app_files.md [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+
+If multiple TIME_INDEPENDENT equations need to be solved, you will need to use conditional statements to calculate the proper residual depending on the elliptic equation being solved. The index of the field being solved can be accessed using the ```this-$>$currentFieldIndex``` statement.
+
+## ICs_and_BCs.cc
+The file ''ICs_and_BCs.cc'' contains the initial conditions for the primary variables as well as the expressions for non-uniform Dirichlet boundary conditions, if present. The file contains two functions: setInitialCondition and setNonUniformDirichletBCs.
+
+### setInitialCondition
+Here's a look at the setInitialCondition function for the coupled Allen-Cahn/Cahn-Hilliard example application:
+```
+template <int dim, int degree>
```
Comment (fractalsbyx) [inline doc/doxygen/user_manual/app_files/app_files.md:231]: do you want to leave in the [[maybe unused]] changes. Or any other template changes i may have missed.

File: doc/doxygen/user_manual/app_files/app_files.md [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+    if (dim ==2) ssf=variable_value(0)-calmin;
+    if (dim ==3) ssf=(variable_value(0)-calmin)*(variable_value(0)-calmin);
+	// Calculate the nucleation rate
+	double J=k1*exp(-k2/(std::max(ssf,1.0e-6)))*exp(-tau/(this->currentTime));
+	double retProb=1.0-exp(-J*userInputs.dtValue*((double)userInputs.steps_between_nucleation_attempts)*dV);
+    return retProb;
+}
+```
+
+## customPDE.h
```
Comment (fractalsbyx) [inline doc/doxygen/user_manual/app_files/app_files.md:433]: I think this should be first

File: doc/doxygen/user_manual/app_files/app_files.md
Comment (fractalsbyx) [inline doc/doxygen/user_manual/app_files/app_files.md:1]: Split into subpages?

File: doc/doxygen/user_manual/input_file/input_file.md [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+
+- DOUBLE: For individual real numbers
+- INT: For individual integers
+- BOOL: For individual booleans (true/false)
+- TENSOR: For either rank 1 tensors (i.e. vectors) or rank 2 tensors (i.e. matrices) with a size of the number of dimensions. These are assumed to be real numbers. Each row should be given by a comma-separated list in parentheses. For example, a 3D vector would be given as ```(2.5, 1.2, 0.1)``` and a 2D matrix would be given as: ```((4.1, 2.0),(1.6, 5.2))```.
+- {[symmetry]} ELASTIC CONSTANTS: For sets of elastic constants, given as a vector. The symmetry options are ISOTROPIC, TRANSVERSE, ORTHOTROPIC, or ANISOTROPIC. The number of entries for the different symmetries are given below. PRISMS-PF converts these sets of elastic constants into a stiffness matrix.
+
+The number and order of the entries for the elastic constants are (where \f$C_{ij}\f$ are entries in the stiffness matrix):
+
+- ISOTROPIC (2D/3D): 2 constants (Young's Modulus, Poisson's Ratio)
```
Comment (fractalsbyx) [inline doc/doxygen/user_manual/input_file/input_file.md:411]: im going to assume this stuff renders properly on the website

**Comment by landinjm** [comment]
These comments can be addressed in the future.

**Pull request merged**